logger.go

```diff
@@ -90,6 +90,14 @@ func (l *logger) SetLevel(logLevel LogLevel) {
 	l.mutLevel.Unlock()
 }
 
+// GetLevel gets the current level of the logger
+func (l *logger) GetLevel() LogLevel {
```

Comment:
Nice implementation. Since we are on this topic, maybe add a new function like 
```GO
GetLogLevel(loggerName string) LogLevel 
```
? this will be useful when having to compute large sets but you do not want to just compute and send it to logger just to find out that the logger dropped the message because it was set on a higher level. What do you think?

Reply:
Added `GetLoggerLogName()` (to avoid name collision).

Also, for what you've said, we can implement this very easily:

```
func  (l *logger) WarnFunc(myFunc function() (string, []interface{})) {
    if ... level is warn {
        message, args := myFunc()
        l.Warn(message, args...)
    }
}
```

Then the caller does this:

```
log.WarnFunc(function() (string, []interface{}) {
    return "diagnostics", []interface{}{ GetStuffComputationallyIntensive() }
})
```

Reply:
Is fine for now. Thanks!